src/screens/BB-BUS/DetailScreen.js

```diff
@@ -57,326 +40,214 @@ const TEXT_BLOCKS_SORTER = {
-
-        // workaround for having camelCase keys in `top10`
-        // GraphQL is returning snake_case, see: https://github.com/d12/graphql-remote_loader/issues/36
-        // transforming method thanks to: https://coderwall.com/p/iprsng/convert-snake-case-to-camelcase
-        data = deepRenameKeys(snake_caseData, (key) =>
-          key.replace(/_\w/g, (m) => m[1].toUpperCase())
-        );
-
-        if (!data || !data.length) return null;
-
-        const { forms, textBlocks, authorities, persons } = data[0];
-        let firstTextBlocks;
-        let sortedTextBlocks;
-
-        if (textBlocks) {
-          sortedTextBlocks = _sortBy(textBlocks, (textBlock) => {
-            // fix for multi nested result form Directus API
-            if (textBlock.textBlock) textBlock = textBlock.textBlock;
-
-            return TEXT_BLOCKS_SORTER[textBlock.name];
-          });
-
-          // filter text blocks we want to render before authorities and persons
-          firstTextBlocks = _remove(sortedTextBlocks, (textBlock) => {
-            // fix for multi nested result form Directus API
-            if (textBlock.textBlock) textBlock = textBlock.textBlock;
-
-            return (
-              textBlock.name.toUpperCase() === 'KURZTEXT' ||
-              textBlock.name.toUpperCase() === 'VOLLTEXT'
-            );
-          });
-
-          // filter text blocks, we do not want to render
-          _remove(sortedTextBlocks, (textBlock) => {
-            // fix for multi nested result form Directus API
-            if (textBlock.textBlock) textBlock = textBlock.textBlock;
-
-            return (
-              textBlock.name.toUpperCase() === 'FACHLICH FREIGEGEBEN DURCH' ||
-              textBlock.name.toUpperCase() === 'FACHLICH FREIGEGEBEN AM'
-            );
-          });
+        refreshControl={
+          <RefreshControl
+            refreshing={refreshing}
+            onRefresh={refresh}
+            colors={[colors.accent]}
+            tintColor={colors.accent}
+          />
         }
-
-        const renderForm = (form) => {
-          // fix for multi nested result form Directus API
-          if (form.forms) form = form.forms;
-
-          const { links } = form;
-
-          return links.map((link) => {
-            // fix for multi nested result form Directus API
-            if (link.links) link = link.links;
-
-            const { id, name, url } = link;
-
-            const openWebScreen = () =>
-              navigation.navigate({
-                name: 'Web',
-                params: {
-                  title: name,
-                  webUrl: url,
-                  rootRouteName
-                }
-              });
-
-            return <Button key={id} title={`${name} online`} onPress={openWebScreen} invert />;
-          });
-        };
-
-        const renderTextBlock = (textBlocks, textBlock, index) => {
-          // fix for multi nested result form Directus API
+      >
+        {!!forms && !!forms.length && (
```

Comment:
we could introduce optional chaining here as well `!!forms?.length`

Reply:
done